Application/Frontend/contexts/queryCacheContext.tsx

```diff
@@ -0,0 +1,61 @@
+import React, { createContext, useContext, useState, ReactNode } from 'react';
+
+// Define an interface for your context state
+interface CacheContextState {
+  lastFetched: number | null;
+  setLastFetched: (value: number | null) => void;
+}
+
+// Define a default context state
+const defaultContextState: CacheContextState = {
+  lastFetched: null,
+  setLastFetched: () => {}, // noop function
```

Comment:
- You might want to add error handling or validation to ensure that the setLastFetched function is only called with valid timestamp values. For example, you could check if the value provided to setLastFetched is a valid number or null.